scriptmodules/supplementary/skyscraper.sh

```diff
@@ -0,0 +1,680 @@
+        [E]="Opens the configuration file \Zbconfig.ini\Zn in an editor."
+        [F]="Opens the artwork definition file \Zbartwork.xml\Zn in an editor."
+        [P]="Purge \ZbALL\Zn all cached resources from the LocalDB."
+        [S]="Purge all cached resources from the LocalDB for a chosen system."
+    )
+
+    while true; do
+
+        db_size=$(du -sh "$configdir/all/skyscraper/dbs" 2>/dev/null | cut -f 1 || echo 0m)
+        [[ -n "$db_size" ]] && db_dize="0Mb"
+
+        local cmd=(dialog --backtitle "$__backtitle" --help-button --colors --no-collapse --default-item "$default" --ok-label "Ok" --cancel-label "Back" --title "Advanced options and commands" --menu "\nLocal cache is currently at $db_size\n\n" 21 65 12)
+
+        local options=("-" "\ZbLocal Cache\Zn")
+
+        if [[ "$cache_screenshots" -eq 1 ]]; then
+            options+=(1 "Cache screenshots (Enabled)")
+        else
+            options+=(1 "Cache screenshots (Disabled)")
+        fi
+
+        if [[ "$cache_covers" -eq 1 ]]; then
+            options+=(2 "Cache covers (Enabled)")
+        else
+            options+=(2 "Cache covers (Disabled)")
+        fi
+
+
+        if [[ "$cache_wheels" -eq 1 ]]; then
+            options+=(3 "Cache wheels (Enabled)")
+        else
+            options+=(3 "Cache wheels (Disabled)")
+        fi
+
+
+        if [[ "$cache_marquees" -eq 1 ]]; then
+            options+=(4 "Cache marquees (Enabled)")
+        else
+            options+=(4 "Cache marquees (Disabled)")
+        fi
+
+        options+=("-" "\ZbPurge commands\Zn")
+        options+=(S "Purge system")
+        options+=(P "Purge all (\Z1!\Z0)")
+
+
+        options+=("-" "\Zb\Z1Expert\Z0 - edit configurations")
+        options+=(E "Edit \Zuconfig.ini\Zn")
+        options+=(F "Edit \Zuartwork.xml\Zn")
+
+        local choice=$("${cmd[@]}" "${options[@]}" 2>&1 > /dev/tty)
+
+        if [[ -n "$choice" ]]; then
+            local default=$choice
+
+            case "$choice" in
+
+                1)
+                    cache_screenshots="$((cache_screenshots ^ 1))"
+                    iniSet "cache_screenshots" "$cache_screenshots"
+                    ;;
+
+                2)
+                    cache_covers="$((cache_covers ^ 1))"
+                    iniSet "cache_covers" "$cache_covers"
+                   ;;
+
+                3)
+                    cache_wheels="$((cache_wheels ^ 1))"
+                    iniSet "cache_wheels" "$cache_wheels"
+                   ;;
+
+                4)
+                    cache_marquees="$((cache_marquees ^ 1))"
+                    iniSet "cache_marquees" "$cache_marquees"
+                   ;;
+
+                S)
+                    _purge_platform_skyscraper
+                    ;;
+
+                P)
+                    confirm=$(dialog --clear --beep  --defaultno --colors --yesno  "\Z1\ZbAre you sure ?\Zn\nThis will \Zb\ZuERASE\Zn all locally cached scraped resources" 8 60 2>&1 >/dev/tty)
```

Comment:
we don't use beep anywhere else. Also we don't use --colors elsewhere, so might be best to stick with the look of other dialogs ?

Reply:
`--beep` is a leftover experiment, but colors are used mainly in the Help section for formatting and only for Bold/Underline. I'll eliminate any formatting on the main screen, but I'd like to keep them in the Help sections.